boards/components/src/hmac.rs

```diff
@@ -0,0 +1,161 @@
+//!
+//! Usage
+//! -----
+//! ```rust
+//!    let hmac_data_buffer = static_init!([u8; 64], [0; 64]);
+//!    let hmac_dest_buffer = static_init!([u8; 32], [0; 32]);
+//!
+//!    let mux_hmac = components::hmac::HmacMuxComponent::new(&ibex::hmac::HMAC).finalize(
+//!        components::hmac_mux_component_helper!(lowrisc::hmac::Hmac, [u8; 32]),
+//!    );
+//!
+//!    let hmac = components::hmac::HmacsComponent::new(
+//!        board_kernel,
+//!        &mux_hmac,
+//!        hmac_data_buffer,
+//!        hmac_dest_buffer,
+//!    )
+//!    .finalize(components::hmac_component_helper!(
+//!        lowrisc::hmac::Hmac,
+//!        [u8; 32]
+//!    ));
+//! ));
+//! ```
+
+use capsules;
+use capsules::hmac::HmacDriver;
+use capsules::virtual_hmac::MuxHmac;
+use capsules::virtual_hmac::VirtualMuxHmac;
+use core::marker::PhantomData;
+use core::mem::MaybeUninit;
+use kernel::capabilities;
+use kernel::component::Component;
+use kernel::create_capability;
+use kernel::hil::digest;
+use kernel::static_init_half;
+
+// Setup static space for the objects.
+#[macro_export]
+macro_rules! hmac_mux_component_helper {
+    ($A:ty, $T:ty) => {{
+        use capsules::virtual_hmac::MuxHmac;
+        use capsules::virtual_hmac::VirtualMuxHmac;
+        use core::mem::MaybeUninit;
+        static mut BUF1: MaybeUninit<MuxHmac<'static, $A, $T>> = MaybeUninit::uninit();
+        &mut BUF1
+    };};
+}
+
+pub struct HmacMuxComponent<
+    A: 'static + digest::Digest<'static, T>,
+    T: 'static + digest::DigestType,
+> {
+    hmac: &'static A,
+    phantom: PhantomData<&'static T>,
+}
+
+impl<A: 'static + digest::Digest<'static, T>, T: 'static + digest::DigestType>
+    HmacMuxComponent<A, T>
+{
+    pub fn new(hmac: &'static A) -> HmacMuxComponent<A, T> {
+        HmacMuxComponent {
+            hmac,
+            phantom: PhantomData,
+        }
+    }
+}
+
+impl<A: 'static + digest::Digest<'static, T>, T: 'static + digest::DigestType> Component
+    for HmacMuxComponent<A, T>
+{
+    type StaticInput = &'static mut MaybeUninit<MuxHmac<'static, A, T>>;
+    type Output = &'static MuxHmac<'static, A, T>;
+
+    unsafe fn finalize(self, s: Self::StaticInput) -> Self::Output {
+        let mux_hmac = static_init_half!(s, MuxHmac<'static, A, T>, MuxHmac::new(self.hmac));
+
+        mux_hmac
+    }
+}
+
+// Setup static space for the objects.
+#[macro_export]
+macro_rules! hmac_component_helper {
+    ($A:ty, $T:ty) => {{
+        use capsules::hmac::HmacDriver;
+        use capsules::virtual_hmac::MuxHmac;
+        use capsules::virtual_hmac::VirtualMuxHmac;
+        use core::mem::MaybeUninit;
+        static mut BUF1: MaybeUninit<VirtualMuxHmac<'static, $A, $T>> = MaybeUninit::uninit();
+        static mut BUF2: MaybeUninit<HmacDriver<'static, VirtualMuxHmac<'static, $A, $T>, $T>> =
+            MaybeUninit::uninit();
+        (&mut BUF1, &mut BUF2)
+    };};
+}
+
+pub struct HmacsComponent<A: 'static + digest::Digest<'static, T>, T: 'static + digest::DigestType>
```

Comment:
Why is this called `HmacsComponent` (plural)?

Reply:
That's what the `LED` one did, I have dropped the s